packages/website/pages/api/preview.ts

```diff
@@ -0,0 +1,33 @@
+import { getSingleArticleBySlug } from '../../lib/api';
+
+export default async function preview(req, res) {
+  const { secret, slug } = req.query;
+
+  if (secret !== process.env.CONTENTFUL_PREVIEW_ACCESS_TOKEN || !slug) {
+    return res.status(401).json({ message: 'Invalid token' });
+  }
+
+  // Fetch the headless CMS to check if the provided `slug` exists
+  const post = await getSingleArticleBySlug(slug, true);
+
+  // If the slug doesn't exist prevent preview mode from being enabled
+  if (!post) {
+    return res.status(401).json({ message: 'Invalid slug' });
+  }
+
+  // Enable Preview Mode by setting the cookies
+  res.setPreviewData({});
+
+  // Redirect to the path from the fetched post
+  // We don't redirect to req.query.slug as that might lead to open redirect vulnerabilities
+  // res.writeHead(307, { Location: `/${post.slug}` })
+  const url = `/${post.slug}`;
+  res.setHeader('Content-Type', 'text/html');
+  res.write(
```

Comment:
I think you can do something like (pseudo code) `res.status(200).redirect(307, url);` instead.